src/build.sh

```diff
@@ -181,6 +187,18 @@ function build_clean()
   cmd_manager "$flag" "$cmd"
 }
 
+# This function runs the 'make distclean' command under the hood, with
+# the advantage of checking if the user is using an env or not.
+# In other words, it integrates env with the distclean option.
+function build_distclean()
+{
+  local flag="$1"
+  local cmd
+
+  cmd="make distclean${output_kbuild_flag}"
```

Comment:
Notice that `output_kbuild_flag` is not global; in this sense, you must receive this as a parameter or call `get_current_env_name`.
Btw... this code should not work inside env, did you check it?

Reply:
I don't understand very well, isn't this implementation supposed to be integrated into the env?  

Actually by my tests it seems that it is working inside the env.

Reply:
Notice that we never declared `output_kbuild_flag` as global (see `declare -g` at the beginning of this file). If your test passes, it may be because of another test setup. Something like this:

1. Sub-test X: Setup env
2. Sub-test X+n: This is your test, and it is passing because Sub-test X created an env, and we never cleaned it.
(this is a bug in the test since each unit test should be independent)

Ensure that the `tearDown` function is in the unit test and that it cleans everything created by the previous test, and try again.

Reply:
Please, address my previous comment.

The solution should be to use the below code inside this function to get the `output_kbuild_flag`.

```bash
  env_name=$(get_current_env_name)
  if [[ "$?" == 0 ]]; then
    options_values['ENV_PATH_KBUILD_OUTPUT_FLAG']="${KW_CACHE_DIR}/${ENV_DIR}/${env_name}"
    output_kbuild_flag=" O=${options_values['ENV_PATH_KBUILD_OUTPUT_FLAG']}"
  fi
```